app.js

```diff
@@ -101,6 +101,45 @@ async function createInstance(action, settings) {
         waitForOperation
     );
 
+    // create firewall rules
+    if (parsers.boolean(action.params.allowHttp) || parsers.boolean(action.params.allowHttps)) {
+
+        const n = net.lastIndexOf('/');
+        const netshortname = net.substring(n + 1);
+
+        let firewallResource = removeUndefinedAndEmpty({
+            name: `${netshortname}-allow-http`,
+            network: net,
+            priority: 1000,
+            destinationRanges: [],
+            sourceRanges: ["0.0.0.0/0"],
+            direction: "INGRESS",
+            targetTags: ["http-server"],
+            allowed: [{
+                IPProtocol: 'tcp',
+                ports: [80]
+            }]
+        });    
+
+        try {
+            const fwHttpResult = await computeClient.createFirewallRule(firewallResource, waitForOperation);
+            } catch (error) {
+            // throw error;
```

Comment:
please uncomment this, I have added a check in createFirewallRule method, if firewall rule with the same name is present it throws an error telling that.